### PR TITLE
Domains: full flow.

### PR DESCRIPTION
**Pull request by jklausa**
This should be the last PR of domains (apart from some UI bugs that I'll fix tomorrow, including better iPad layout and what @ScoutHarris noticed in the last PR Review).

This should allow you to redeem a domain and then fall back to a plugin install process seemlessly — allowing Biz users that don't have custom domains to complete the entire flow.

Please not that due to the API (and it's flakiness — I've basically added "oh this failed? that doesn't seem right. lets retry 10 times" logic in multiple places)  this process can take _excruciatingly_ long time — up to 5 minutes wall time. 

To test:

1. Get a Biz domain without domain (ping me if you need credits or to cancel a domain, I have SA superpowers).
2. Try to install a plugin
3. If in doubt, watch console output to see whether everything is going fine
4. Verify that the plugin installed correctly

### CODE REVIEW

**Comment by jklausa** [comment]
(disregard the conflicts for now: I'll get them resolved when #10454 gets merged back to develop and I'll update what branch this PR is against to `develop`)

**Comment by jklausa** [comment]
Conflicts resolved, branch is now pointing at `develop` — this one's ready to go :)

File: WordPress/Classes/Utility/Automated Transfer/AutomatedTransferHelper.swift [hunk 1]
```diff
@@ -128,12 +167,54 @@ class AutomatedTransferHelper {
             }
 
         }, failure: { (error) in
-            DDLogInfo(("[AT] Failed to initiate Automated Transfer: \(error)"))
 
-            WPAnalytics.track(.automatedTransferInitiationFailed)
+            // The async nature of AT process bites us here again. Sometimes, even though the backend says
+            // 'hell yeah son, everything's fine, go ahead!' in the eligibility check, trying to actually
+            // start the AT process (usually only after a custom domain was _just_ purchased) fails spectacularly.
+            // In those ceases, we'll retry this request a few times.
```
Comment (aerych) [inline WordPress/Classes/Utility/Automated Transfer/AutomatedTransferHelper.swift]: ceases => cases

**Comment by aerych** [comment]
Heya @jklausa!  I took the branch for a quick spin (I chose the yoast seo plugin).  The domain flow went very fast, the plugin install took ages, but everything worked! 🎉 

One thing I noticed,  while the domain is being processed we show a HUD spinner. While its visible the button to finish the domain purchase is still tappable.  I was able to tap multiple times and got an error that there was a problem completing the purchase (sorry, I didn't grab a screen shot).  I think the error was for one of the subsequent taps because the domain purchase succeeded regardless.   Should we disable the button while the request is being processed and the HUD spinner is visible?

**Comment by jklausa** [comment]
Ah, great catch! I’ll get right on that. 

And yeah - domain going fast and AT taking foreverrrrrrrrrrrrrr is what I’ve been seeing too.  Not much we can do on our end to help that :(

**Comment by jklausa** [comment]
@aerych Added the code to disable the button while request is in flight and fixed the typo :)

**Comment by jklausa** [comment]
oh my god this is finally happening

**Pull request merged**